modules/core/src/test/java/org/apache/ignite/spi/communication/tcp/GridTcpCommunicationSpiConfigSelfTest.java

```diff
@@ -70,4 +98,67 @@ public void testLocalPortRange() throws Exception {
         startGrid(cfg);
     }
 
+    /**
+     * Test checks that {@link TcpCommunicationSpi#ATTR_HOST_NAMES} attribute
+     * is empty only if {@link
+     * IgniteSystemProperties#IGNITE_TCP_COMM_SET_ATTR_HOST_NAMES} ==
+     * {@code false} and local host is set via {@link
+     * IgniteConfiguration#setLocalHost}.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    @WithSystemProperty(key = IGNITE_TCP_COMM_SET_ATTR_HOST_NAMES, value = "false")
```

Comment:
Why is default value need to define explicitly?

Reply:
To save default behavior, but may need to divide it into 2 tests.